src/background/providers/index.ts

```diff
@@ -13,4 +13,10 @@ export interface Provider {
     handleBeforeSendHeaders(
         details: chrome.webRequest.WebRequestHeadersDetails,
     ): chrome.webRequest.BlockingResponse | void;
+    handleActivated(): void;
+}
+
+export interface Callbacks {
```

Comment:
is it possible to move the methods in `Callbacks` to the `Provider` interface?

Reply:
`Callbacks` is supposed to be used in constructors but there is no way to define the constructor signatures in the `Provider` interface.